include/alpaka/kernel/TaskKernelCpuOmp2Blocks.hpp

```diff
@@ -156,15 +170,38 @@ namespace alpaka
                 *static_cast<WorkDivMembers<TDim, TIdx> const*>(this),
                 blockSharedMemDynSizeBytes);
 
-            // NOTE: schedule(static) does not improve performance.
+            //! Set the given OpenMP schedule while this object is alive.
+            //! Restore the old schedule afterwards.
+            class ScheduleGuard
+            {
+            public:
+                ScheduleGuard(omp::Schedule const schedule) : oldSchedule(omp::getSchedule())
+                {
+                    omp::setSchedule(schedule);
+                }
+
+                ScheduleGuard(ScheduleGuard const&) = default;
+
+                ~ScheduleGuard()
+                {
+                    omp::setSchedule(oldSchedule);
+                }
+
+            private:
+                omp::Schedule const oldSchedule;
+            };
+
+            // Schedule change is a scoped object, so that the old schedule is
+            // also restored in case of exception
+            auto const scheduleGuard = ScheduleGuard{schedule};
 #    if _OPENMP < 200805 // For OpenMP < 3.0 you have to declare the loop index (a signed integer) outside of the loop
                          // header.
             std::intmax_t iNumBlocksInGrid(static_cast<std::intmax_t>(numBlocksInGrid));
             std::intmax_t i;
-#        pragma omp for nowait schedule(guided)
+#        pragma omp for nowait schedule(runtime)
             for(i = 0; i < iNumBlocksInGrid; ++i)
 #    else
-#        pragma omp for nowait schedule(guided)
+#        pragma omp for nowait schedule(runtime)
             for(TIdx i = 0; i < numBlocksInGrid; ++i)
```

Comment:
Note: with thie PR, the Omp2Blocks accelerator will always use `runtime`. For the implementations that only has 2.0 support (so for Microsoft) this means the default changes from guided to whatever the default for the environment variable. I think this is okay, since for 3.0 and higher the default will stay guided as it was.